core/src/block.rs

```diff
@@ -21,13 +21,21 @@ use parity_scale_codec::{Decode, Encode};
 use crate::{
     prelude::*,
     sumeragi::{
+        config::*,
         network_topology::Topology,
         view_change::{Proof, ProofChain as ViewChangeProofs},
     },
     tx::{TransactionValidator, VersionedAcceptedTransaction},
     wsv::WorldTrait,
 };
 
+const PIPELINE_TIME_MS: u64 =
+    DEFAULT_BLOCK_TIME_MS + DEFAULT_COMMIT_TIME_MS + DEFAULT_TX_RECEIPT_TIME_MS;
+
+/// Default estimation of consensus duration
+#[allow(clippy::integer_division)]
+pub const DEFAULT_CONSENSUS_ESTIMATION: u64 = (DEFAULT_COMMIT_TIME_MS + PIPELINE_TIME_MS) / 2;
```

Comment:
```suggestion
/// Default estimation of consensus duration
pub const DEFAULT_CONSENSUS_ESTIMATION_MS: u64 = DEFAULT_TX_RECEIPT_TIME_MS + DEFAULT_BLOCK_TIME_MS + DEFAULT_COMMIT_TIME_MS;
```
Why do we give special treatment to `DEFAULT_COMMIT_TIME_MS` -- the phase waiting for the proxy tail?

Reply:
I thought we decided to use the average of commit time and pipeline time?

Reply:
I may have missed the point in our last discussion. Could you provide an explanation @appetrosyan ?

Reply:
Ah, I see. This represents:
The gap between `PendingBlock::new()` and the estimated actual commit = `(best + worst) / 2`

Reply:
@s8sato , I'm guessing that this information should probably be included as part of the documentation. I've added a few lines about this to the tutorial repository, PR pending. 

In  the meantime. 

While the estimate should be as accurate as we can get it, the logic isn't `(best+worst)/2`. The end of consensus is not a well-defined concept. 

Internally, when the block commits is when the next cycle begins. However, if you submit at that time, it's likely that the system won't respond accurately. 

Externally the pipeline time is what acts as a cut-off point. After waiting for `PIPELINE` time, you can submit queries/ISI and get a valid answer. 

Since both are equally valid choices, what I said during our conversation was "I don't care which one you pick, just pick one. It can even be the average, if choosing between the two is too difficult". That's all.